notebooks/246-paint-by-example/246-paint-by-example.ipynb

```diff
@@ -0,0 +1,943 @@
+    "            image (List of np.ndarray or PIL.Image.Image):\n",
+    "                Postprocessed images\n",
+    "        \"\"\"\n",
+    "        if \"padding\" in meta:\n",
+    "            pad = meta[\"padding\"]\n",
+    "            (_, end_h), (_, end_w) = pad[1:3]\n",
+    "            h, w = image.shape[2:]\n",
+    "            unpad_h = h - end_h\n",
+    "            unpad_w = w - end_w\n",
+    "            image = image[:, :, :unpad_h, :unpad_w]\n",
+    "        image = np.clip(image / 2 + 0.5, 0, 1)\n",
+    "        image = np.transpose(image, (0, 2, 3, 1))\n",
+    "        # 9. Convert to PIL\n",
+    "        if output_type == \"pil\":\n",
+    "            image = self.numpy_to_pil(image)\n",
+    "            if \"src_height\" in meta:\n",
+    "                orig_height, orig_width = meta[\"src_height\"], meta[\"src_width\"]\n",
+    "                image = [img.resize((orig_width, orig_height),\n",
+    "                                    PIL.Image.Resampling.LANCZOS) for img in image]\n",
+    "        else:\n",
+    "            if \"src_height\" in meta:\n",
+    "                orig_height, orig_width = meta[\"src_height\"], meta[\"src_width\"]\n",
+    "                image = [cv2.resize(img, (orig_width, orig_width))\n",
+    "                         for img in image]\n",
+    "        return image\n",
+    "\n",
+    "    def get_timesteps(self, num_inference_steps:int, strength:float):\n",
+    "        \"\"\"\n",
+    "        Helper function for getting scheduler timesteps for generation\n",
+    "        In case of image-to-image generation, it updates number of steps according to strength\n",
+    "        \n",
+    "        Parameters:\n",
+    "           num_inference_steps (int):\n",
+    "              number of inference steps for generation\n",
+    "           strength (float):\n",
+    "               value between 0.0 and 1.0, that controls the amount of noise that is added to the input image. \n",
+    "               Values that approach 1.0 allow for lots of variations but will also produce images that are not semantically consistent with the input.\n",
+    "        \"\"\"\n",
+    "        # get the original timestep using init_timestep\n",
+    "        init_timestep = min(int(num_inference_steps * strength), num_inference_steps)\n",
+    "\n",
+    "        t_start = max(num_inference_steps - init_timestep, 0)\n",
+    "        timesteps = self.scheduler.timesteps[t_start:]\n",
+    "\n",
+    "        return timesteps, num_inference_steps - t_start "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure Inference Pipeline\n",
+    "\n",
+    "Configuration steps:\n",
+    "1. Load models on device\n",
+    "2. Configure tokenizer and scheduler\n",
+    "3. Create instance of OvStableDiffusionInpaintingPipeline class\n",
+    "\n",
+    "This can take a while to run."
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
It's recommended to include a feature that allows users to select the inference device via a dropdown menu. This can be implemented using `ipywidgets`, enabling a convenient and interactive way to choose from available devices, including an "AUTO" option for automatic device selection. Please take a look at the 236 notebook.